client/src/templates/Challenges/generic/show.tsx

```diff
@@ -0,0 +1,260 @@
+import { graphql } from 'gatsby';
+import React, { useEffect, useRef, useState } from 'react';
+import Helmet from 'react-helmet';
+import { useTranslation } from 'react-i18next';
+import { connect } from 'react-redux';
+import { Container, Col, Row, Button } from '@freecodecamp/ui';
+
+// Local Utilities
+import Spacer from '../../../components/helpers/spacer';
+import LearnLayout from '../../../components/layouts/learn';
+import { ChallengeNode, ChallengeMeta, Test } from '../../../redux/prop-types';
+import ChallengeDescription from '../components/challenge-description';
+import Hotkeys from '../components/hotkeys';
+import ChallengeTitle from '../components/challenge-title';
+import VideoPlayer from '../components/video-player';
+import CompletionModal from '../components/completion-modal';
+import Assignments from '../components/assignments';
+import {
+  challengeMounted,
+  updateChallengeMeta,
+  openModal,
+  updateSolutionFormValues,
+  initTests
+} from '../redux/actions';
+import { isChallengeCompletedSelector } from '../redux/selectors';
+import { BlockTypes } from '../../../../../shared/config/blocks';
+
+// Redux Setup
+const mapStateToProps = (state: unknown) => ({
+  isChallengeCompleted: isChallengeCompletedSelector(state) as boolean
+});
+
+const mapDispatchToProps = {
+  initTests,
+  updateChallengeMeta,
+  challengeMounted,
+  updateSolutionFormValues,
+  openCompletionModal: () => openModal('completion')
+};
+
+// Types
+interface ShowQuizProps {
+  challengeMounted: (arg0: string) => void;
+  data: { challengeNode: ChallengeNode };
+  description: string;
+  initTests: (xs: Test[]) => void;
+  isChallengeCompleted: boolean;
+  openCompletionModal: () => void;
+  pageContext: {
+    challengeMeta: ChallengeMeta;
+  };
+  updateChallengeMeta: (arg0: ChallengeMeta) => void;
+  updateSolutionFormValues: () => void;
+}
+
+const ShowGeneric = ({
+  challengeMounted,
+  data: {
+    challengeNode: {
+      challenge: {
+        assignments,
+        bilibiliIds,
+        block,
+        blockType,
+        description,
+        challengeType,
+        fields: { tests },
+        helpCategory,
+        instructions,
+        title,
+        translationPending,
+        superBlock,
+        videoId,
+        videoLocaleIds
+      }
+    }
+  },
+  pageContext: { challengeMeta },
+  initTests,
+  updateChallengeMeta,
+  openCompletionModal,
+  isChallengeCompleted
+}: ShowQuizProps) => {
+  const { t } = useTranslation();
+  const { nextChallengePath, prevChallengePath } = challengeMeta;
+  const container = useRef<HTMLElement | null>(null);
+
+  const blockNameTitle = `${t(
+    `intro:${superBlock}.blocks.${block}.title`
+  )} - ${title}`;
+
+  useEffect(() => {
+    initTests(tests);
```

Comment:
Discussed a bit with Oliver.

Ideally, this should be handled by the component needing the tests. However, because not all components clean up old tests by themselves, this has to be done on any page change.

Ideally, whichever component needs tests also has a `useEffect(() => { return () => initTests([]) }, [])` cleanup function.

Until then, removing this here might have buggy consequences.